test/e2e/snaps/test-snap-update-component.spec.js

```diff
@@ -0,0 +1,156 @@
+
+describe('Test Snap update via snaps component', function () {
+  it('can install an old and then update via the snaps component', async function () {
+    const ganacheOptions = {
+      accounts: [
+        {
+          secretKey:
+            '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+          balance: 25000000000000000000,
+        },
+      ],
+    };
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions,
+        failOnConsoleError: false,
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+
+        // open a new tab and navigate to test snaps page and connect
+        await driver.openNewPage(TEST_SNAPS_WEBSITE_URL);
+        await driver.delay(1000);
+
+        // find and scroll to the correct card and click first
+        const snapButton = await driver.findElement('#connectUpdate');
+        await driver.scrollToElement(snapButton);
+        await driver.delay(1000);
+        await driver.clickElement('#connectUpdate');
+        await driver.delay(1000);
+
+        // switch to metamask extension and click connect
+        await switchToNotificationWindow(driver, 3);
+        await driver.clickElement({
+          text: 'Connect',
+          tag: 'button',
+        });
+
+        await driver.waitForSelector({ text: 'Install' });
+
+        await driver.clickElementSafe('[data-testid="snap-install-scroll"]');
+
+        await driver.clickElement({
+          text: 'Install',
+          tag: 'button',
+        });
+
+        // wait for permissions popover, click checkboxes and confirm
+        await driver.delay(500);
+        await driver.clickElement('.mm-checkbox__input');
+        await driver.clickElement({
+          text: 'Confirm',
+          tag: 'button',
+        });
+
+        await driver.waitForSelector({ text: 'OK' });
+
+        await driver.clickElement({
+          text: 'OK',
+          tag: 'button',
+        });
+
+        // navigate to test snap page
+        const windowHandles = await driver.waitUntilXWindowHandles(
+          2,
+          1000,
+          10000,
+        );
+        await driver.switchToWindow(windowHandles[1]);
+
+        // wait for npm installation success
+        await driver.waitForSelector({
+          css: '#connectUpdate',
+          text: 'Reconnect to Update Snap',
+        });
+
+        // switch to the original MM tab
+        const extensionPage = windowHandles[0];
+        await driver.switchToWindow(extensionPage);
+        await driver.delay(1000);
+
+        // click on the global action menu
+        await driver.clickElement(
+          '[data-testid="account-options-menu-button"]',
+        );
+
+        // try to click on the snaps item
+        await driver.clickElement({
+          text: 'Snaps',
+          tag: 'div',
+        });
+        await driver.delay(1000);
```

Comment:
Can we try to reduce number of `delays` used by using `waitForSelector`? I'm not sure if this one particular delay is required since it's in the same window/popup.